File: webserver.js
var http = require('http');

console.log('=== server listening on 127.0.0.1:8080 ===');

http.createServer(function(req, res) {
  // json post request
  if (req.url.indexOf('json') >= 0 && req.method === 'POST') {
    var body = '';
    req.on('data', function(data) {
      body += data;
    });
    req.on('end', function() {
      res.writeHead(200, {'Content-Type': 'application/json'});
      res.end(JSON.stringify(JSON.parse(body)));
    });
  }
  // hello world get request 
  else if (req.url.indexOf('hello') >= 0) {
    res.writeHead(200, {'Content-Type': 'text/plain'});
    res.end('Hello World\n');
  }
}).listen(8080, function() {
  console.log('begin benchmark');         
});

File: go/stdlib/webserver.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"time"
)

func handleHello(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "text/plain")
	fmt.Fprint(w, "Hello World")
}

func handleJson(w http.ResponseWriter, r *http.Request) {
	decoder := json.NewDecoder(r.Body)
	var j []map[string]interface{}
	err := decoder.Decode(&j)
	if err != nil {
		panic(err)
	}
	jstr, err := json.Marshal(j)
	if err != nil {
		panic(err)
	}
	w.Header().Set("Content-Type", "application/json")
	fmt.Fprint(w, string(jstr))
}

func handleDelay(w http.ResponseWriter, r *http.Request) {
	time.Sleep(100 * time.Millisecond)
	handleHello(w, r)
}

func main() {
	http.HandleFunc("/hello", handleHello)
	http.HandleFunc("/json", handleJson)
	http.HandleFunc("/delay", handleDelay)

	fmt.Println("=== server running on 127.0.0.1:8080 ===")
	fmt.Println("begin benchmark")
	fmt.Println(http.ListenAndServe(":8080", nil))
}
